src/components/chips/js/chipsDirective.js

```diff
@@ -0,0 +1,86 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.chips')
+      .directive('mdChips', MdChips);
+
+  /**
+   * @ngdoc directive
+   * @name mdChips
+   * @module material.components.chips
+   *
+   * @description
+   * `<md-chips>` 
+   *
+   */
+
+  function MdChips ($compile) {
+    return {
+      template: function(element, attrs) {
+        var userTemplate = element.clone();
+        attrs['$mdUserTemplate'] = userTemplate;
+        return '\
+          <md-chips-wrap ng-class="{readonly : $mdChipsCtrl.readonly}">\
+          <ul role="presentation">\
+            <li ng-repeat="(index, $chip) in $mdChipsCtrl.items"\
+                class="md-chip-list-item">\
+            </li>\
+            <li ng-if="!$mdChipsCtrl.readonly" class="md-chip-worker"></li>\
+          </ul>\
+          </md-chips-wrap>';
+      },
+      require: ['ngModel', 'mdChips'],
+      controller:   'MdChipsCtrl',
+      controllerAs: '$mdChipsCtrl',
+      bindToController: true,
+      compile: compile,
+      scope: {
+        readonly: '=readonly'
+      }
+    };
+    function compile(element, attr) {
+      var userTemplate = attr['$mdUserTemplate'];
+      var chipEl = userTemplate.find('md-chip');
```

Comment:
I suggest just working with raw DOM elements here instead of going through `angular.element`. The DOM APIs are quite usable for the browsers we support today.